From b63d2f62b6520c6013ede9a18ff2bf3e40fa1f93 Mon Sep 17 00:00:00 2001
Message-Id: <b63d2f62b6520c6013ede9a18ff2bf3e40fa1f93.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Akash Goel <akash.goel@intel.com>
Date: Mon, 23 Dec 2013 12:50:06 +0530
Subject: [PATCH 120/312] FOR_UPSTREAM [VPG]: drm/i915: Added an identifier
 for User frame buffers to Gem obj structure.

In order to uniquely identify the GEM objects representing the User
created frame buffers, a new bit-field 'user_fb' is added to the
GEM obj structure. The User created frame buffers are suitable for
allocation from stolen area. This bit field will be used when
allocating the backing physical space for the GEM objects, so
if the physical space is being allocated for a User frame buffer
the driver could use the free space from the stolen area instead
of shmem.

Issue: VIZ-3430
Change-Id: Ie7d926454232564c7c2a28ed78965ba7bfab2b8a
Signed-off-by: Akash Goel <akash.goel@intel.com>
Signed-off-by: Deepak S <deepak.s@intel.com>
Reviewed-on: https://android.intel.com/164327
Tested-by: sys_abtbuild <sys_abtbuild@intel.com>
Reviewed-by: Bloomfield, Jon <jon.bloomfield@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h      |    5 +++++
 drivers/gpu/drm/i915/intel_display.c |    2 ++
 2 files changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 9d99581..5ce0a01 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -1800,6 +1800,11 @@ struct drm_i915_gem_object {
 	unsigned int has_global_gtt_mapping:1;
 	unsigned int has_dma_mapping:1;
 
+	/*
+	 * Is the object associated with user created FB
+	 */
+	unsigned int user_fb:1;
+
 	struct sg_table *pages;
 	int pages_pin_count;
 
diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 7b0c343..95741ba 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -12409,6 +12409,7 @@ static void intel_user_framebuffer_destroy(struct drm_framebuffer *fb)
 {
 	struct intel_framebuffer *intel_fb = to_intel_framebuffer(fb);
 
+	intel_fb->obj->user_fb = 0;
 	drm_framebuffer_cleanup(fb);
 	WARN_ON(!intel_fb->obj->framebuffer_references--);
 	drm_gem_object_unreference_unlocked(&intel_fb->obj->base);
@@ -12560,6 +12561,7 @@ intel_user_framebuffer_create(struct drm_device *dev,
 	if (&obj->base == NULL)
 		return ERR_PTR(-ENOENT);
 
+	obj->user_fb = 1;
 	return intel_framebuffer_create(dev, mode_cmd, obj);
 }
 
-- 
1.7.9.5

